src/test/scala/io/citrine/lolo/DataGenerator.scala

```diff
@@ -0,0 +1,113 @@
+package io.citrine.lolo
+
+import breeze.stats.variance
+import io.citrine.lolo.linear.LinearRegressionLearner
+import io.citrine.lolo.stats.functions.Friedman
+import io.citrine.random.Random
+
+object DataGenerator {
+
+  /** Sequence of inputs and labels generated for training a learner. */
+  case class TrainingData[I, L](data: Seq[(Vector[I], L)])
```

Comment:
Putting a type parameter on the input doesn't seem like it gains us anything

Reply:
There is one test that relied on the fact that the initial inputs are a `Vector[Double]`. But I'm fine just casting for that one test for the sake of a cleaner trait.

Reply:
I lied. There's are actually a few tests that rely on the fact that `DataGenerator.generate` initially produces continuous data of type `Seq[(Vector[Double], Double)]`. If we remove the type parameter `I` on the inputs, then we need to store in the training data class a `Seq[(Vector[Any], L)]` and resort to casting for when the inputs are required to be `Double`...which partially defeats the purpose of this interface.

Reply:
@sfriedowitz, that makes sense to me. Is this something that you would come back to at a later date (because the additional code changes are out of scope), or is this just the best way for writing the code? Maybe @bfolie has some additional thoughts on this?

Reply:
Why do the tests require `Vector[Double]`? Are there any instances in the main code for which we want to restrict the input type? Making the interface more complex _solely_ to satisfy a test makes me suspicious.

Reply:
> Are there any instances in the main code for which we want to restrict the input type

This interface is solely used as a data generator for test code. It never appears in the package source.

> Why do the tests require Vector[Double]?

A variety of reasons. A couple of tests jitter the data surrounding an input feature by adding `+/-` some noise. Another feeds an input feature into one of the Friedman functions. And many tests simply "bin" the input features, which requires them to be of type double.

Reply:
The test code is littered with the following steps:
- Generate continuous input data of type Seq[(Vector[Double], Double)]`
- Transform either the input features or label values in some way
- Train a model on the transformed data, which must conform to the learner input type of `Seq[(Vector[Any], T)]`.

A generic on the input data is to allow for transformations in step 2 that operate on continuous (type `Double)` input data. Not because the `Learner[T]` interface is unable to handle data with `Vector[Any]` inputs.

Reply:
> This interface is solely used as a data generator for test code. It never appears in the package source.

Alright. As long as it's confined to the test code and doesn't become the interface for training data rows (for which I assume we'll want to have a label type parameter) then I'm OK with this